### ISSUE TITLE
extracting subdirectory requires directory entries in the zip

### ISSUE THREAD

**Issue by peteruhnak** [srikanth-lingala/zip4j#347]
According to the docs, it should be possible to extract subdirectories, e.g.

```
new ZipFile("filename.zip").extractFile("folderNameInZip/", "/destination_directory");
```

However, this requires a directory entry to be present in the archive:

* https://github.com/srikanth-lingala/zip4j/blob/v2.9.0/src/main/java/net/lingala/zip4j/ZipFile.java#L638-L640
* https://github.com/srikanth-lingala/zip4j/blob/v2.9.0/src/main/java/net/lingala/zip4j/headers/HeaderUtil.java#L69-L79

As far as I understand, having directory entries in zips is optional, so the unzipping should not check for them. Presumably it should only compare against the user-supplied entry name (and ensure the user-supplied name ends with a slash).

Here's an example archive with no directory entires:

[archive-no-dirs.zip](https://github.com/srikanth-lingala/zip4j/files/6973121/archive-no-dirs.zip)

The archive was created with Powershell 7 `Compress-Archive` cmdlet ( https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/compress-archive?view=powershell-7.1 )

**Comment by srikanth-lingala** [comment]
Yes, you are right. It should be possible to extract subdirectories without the subdirectory entry being present. I will fix this.

**Comment by srikanth-lingala** [comment]
Fixed in v2.9.1 released today.
